docs/migration-guide.md

```diff
@@ -0,0 +1,330 @@
+2. Move the imports from the old `*.visual-diff.html` file to the top of the new JavaScript file. Do not import `@brightspace-ui/core`'s `typography.js` as it will be included automatically.
+3. Add `import { expect, fixture, html } from '@brightspace-ui/testing';`
+4. Move the defined fixtures from the HTML file to fixtures in the JavaScript file. Remove `id`s which were used solely for identifying the fixture for `getRect`.
+5. Delete the old `*.visual-diff.html` file
+6. Remove all the old Puppeteer, browser and VisualDiff setup code
+7. Replace `async function() {` with `async () => {`
+8. Rewrite everything using `page.$eval` to access the element directly (just like unit tests do)
+9. Replace calls with `getRect` and `screenshotAndCompare` with `await expect(elem).to.be.golden()`
+10. Make use of the `focusElem`, `clickElem`, `sendKeysElem`, `oneEvent`, etc. helpers where possible
+
+Before (HTML):
+
+```html
+<html>
+  <head>
+    <script type="module">
+      import '@brightspace-ui/core/components/typography/typography.js';
+      import '../my-elem.js';
+    </script>
+  </head>
+  <body class="d2l-typography">
+    <div>
+      <my-elem id="default"></my-elem>
+    </div>
+  </body>
+</html>
+```
+
+Before (JavaScript):
+
+```javascript
+import puppeteer from 'puppeteer';
+import { VisualDiff } from '@brightspace-ui/visual-diff';
+
+describe('my-elem', () => {
+
+  const visualDiff = new VisualDiff('my-elem', import.meta.url);
+
+  let browser, page;
+
+  before(async() => {
+    browser = await puppeteer.launch();
+    page = await visualDiff.createPage(browser);
+    await page.goto(`${visualDiff.getBaseUrl()}/test/my-elem.visual-diff.html`, { waitUntil: ['networkidle0', 'load'] });
+    await page.bringToFront();
+  });
+
+  beforeEach(async() => {
+    await visualDiff.resetFocus(page);
+  });
+
+  after(async() => await browser.close());
+
+  it('passes visual-diff comparison', async function() {
+    await page.$eval('#default', (elem) => {
+      const innerElem = elem.shadowRoot.querySelector('button');
+      button.focus();
+    });
+    const rect = await visualDiff.getRect(page, '#default');
+    await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
+  });
+
+});
+```
+
+After:
+
+```javascript
+import '../my-elem.js'
+import { expect, fixture, focusElem, html } from '@brightspace-ui/testing';
+
+describe('my-elem', () => {
+
+  it('passes visual-diff comparison', async() => {
+    const elem = await fixture(html`<my-elem></my-elem>`);
+    await focusElem(elem.shadowRoot.querySelector('button'));
+    await expect(elem).to.be.golden();
+  });
+
+});
+```
+
+Much less code, right?
+
+### Step 3: Test Migration Locally
+
+Test the migration by running `d2l-test-runner vdiff`, filtering by file using `--filter <file-name>` if appropriate.
+
+If tests fail, run `d2l-test-runner report` to view a HTML report visualizing the differences.
```

Comment:
```suggestion
If tests fail, run `d2l-test-runner report` to view an HTML report visualizing the differences.
```
????

Reply:
Yep you're correct -- if the first sound is a vowel, it's "an".

The "FRA" case is more subjective and depends if you pronounce it "eff are eh" ("an FRA") or "frah" ("a FRA").

Reply:
Yeah, wasn't sure if there was some alternate way to say HTML ("het mal") I'd never heard of 😛